Add Whip Proficiencies and Practice Recipe

epenalver opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully.
Once a pull request is submitted, automatic stylistic and consistency checks will be performed on the PR's changes.
The results can be either seen under the "Files changed" section of a PR or in the check's details.

Rules for suggested pull requests:
- If possible, limit yourself to small changes, 500 strings at max. Exceptions are adding or changing maps, and changes, that won't work unless they are done in a single run (even then there can be ways) - violating it puts a lot of unnecessary work on our merge team.
- Do not scope creep. If you make a pull request "Add new gun", please do not make anything more than adding the gun and following changes, like changing the stats of the gun, removing other guns from itemgroups or tweaking zombie horse stats - violating it makes future search and debugging stuff much harder, since PR name is not related to what is changed in the game. "Who the hell removed the quest item from drop in location X in PR, that adds a new plushie" - this may be a quote from a person who was affected by scope creep
- Do not make omnibus PRs. Meaning do not make a single PR, that fixes ten different, not related issues, at once, even if they are all one string - same as previously mentioned scope creep, it doesn't help to search the changes when debugging, despite all power of git blame tool

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Content "Whip Weapon Proficiencies"

#### Purpose of change
The game does not currently include the ability to gain proficiency with whip-style weapons. While they are admittedly impractical weapons, someone who uses them on a regular basis should be able to get better at using them. (This will also support the use of more effective whips available in mods like Magiclysm or XE)


#### Describe the solution

This PR adds Familiarity, Proficiency, and Mastery for weapons in the WHIP group. Supporting changes for this include the addition of the WHIP group to the list of weapon categories, the addition of a "whips" list of weapons as a crafting requirement, a practice recipe for Whip Familiarity, and the addition of the WHIP flag to the dressage whip item. I'm open to the addition of practice recipes for higher skill levels but I didn't see much sign of current whip practitioners doing anything other than swinging a whip around.

#### Describe alternatives you've considered

Make no changes and leave whips on the sideline.

#### Testing

Launched the game with the proposed changes, was able to access the practice recipe for whips (beginner) and learn the proficiency.

#### Additional context

![image](https://github.com/user-attachments/assets/48d623e6-7190-4392-abd7-3f70bcbae8cb)
![image](https://github.com/user-attachments/assets/3d55052a-1c59-4b3f-80dc-d9ee0d09eb25)


<!--README: Cataclysm: Dark Days Ahead is released under the Creative Commons Attribution ShareAlike 3.0 license.
The code and content of the game are free to use, modify, and redistribute for any purpose whatsoever.
By contributing to the project you agree to the terms of the license and that any contribution you make will also be covered by the same license.
See http://creativecommons.org/licenses/by-sa/3.0/ for details. -->

epenalver committed:
Update melee_weapons.json

Add familiarity, proficiency, and mastery for whips to melee weapon proficiencies.

epenalver committed:
Update weapon_categories.json

Add WHIPS as a weapon category.

epenalver committed:
Update melee.json

Add beginner-level whip practice recipe. I can't find much evidence of a higher tier of whip practice than "swing a whip around", so I'm not adding more advanced recipes.

epenalver committed:
Update melee.json

Add whips as a melee weapon group.

epenalver committed:
Update misc.json

Add the WHIP flag to the dressage whip.

epenalver committed:
Update data/json/requirements/melee.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

epenalver committed:
Update data/json/recipes/practice/melee.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

epenalver committed:
Update data/json/recipes/practice/melee.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

epenalver committed:
Update data/json/proficiencies/melee_weapons.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

epenalver committed:
Update melee_weapons.json

Daved27hundred commented:
give the archeologist profession the proficiency.

epenalver commented:
Honestly, archeologist having whip proficiency was something that occurred to me but I didn't go digging in the professions list to see what was applicable. I think other possible candidates could be mounted police officer, lost submissive (more apt for a dom but idk if we have one), rancher, and cowboy. Rancher, Cowboy and Archeologist definitely make the most sense given that they start with a bullwhip.

epenalver committed:
Update misc.json

add WHIPS weapon_category to dressage whip

epenalver committed:
Update misc.json

Add WHIPS weapon_category to bullwhip and scourge.

Holli-Git commented:
> Honestly, archeologist having whip proficiency was something that occurred to me but I didn't go digging in the professions list to see what was applicable. I think other possible candidates could be mounted police officer, lost submissive (more apt for a dom but idk if we have one), rancher, and cowboy. Rancher, Cowboy and Archeologist definitely make the most sense given that they start with a bullwhip.

Please don't give it to the archeologist, it makes no sense.

Holli-Git commented:
> > Please don't give it to the archeologist, it makes no sense.
> 
> No objections here. Should archeologist start with a bullwhip still?

No

ashGlaw commented:
This is great for my Catavania ideas ♥️

epenalver commented:
Also I'm thinking I'll add whip proficiency intermediate for cowboy and rancher and probably not touch archeologist in this PR.

epenalver committed:
Update melee_weapons.json

Swapped whip proficiency and mastery descriptions and emphasized the wording on mastery.

epenalver committed:
Update melee.json

Add intermediate whip practice recipe which uses candles and lighters to practice accuracy.

epenalver committed:
Update professions.json

Add Whip Familiarity and Whip Proficiency to the Rancher and Cowboy professions.
